src/client/remote/commands.ts

```diff
@@ -48,34 +49,34 @@ export class CommandRegistry implements IExtensionSingleActivationService {
             this.commandManager.registerCommand('jupyter.server.file.new', (item) => this.createNew(item, 'file'))
         );
         this.disposables.push(
-            this.commandManager.registerCommand('jupyter.server.notebook.new', (item) =>
-                this.createNew(item, 'notebook')
-            )
+            this.commandManager.registerCommand('jupyter.server.notebook.new', async (item) => {
+                await this.createNew(item, 'notebook');
+            })
         );
         this.disposables.push(
             this.commandManager.registerCommand('jupyter.server.file.delete', (item) => this.delete(item))
         );
     }
-    private async createNew(item: FileSystemNode | DirectoryNode, type: 'file' | 'directory' | 'notebook') {
-        const remoteFs = await this.fsFactory.getOrCreateRemoteFileSystem(item.data.connection);
-        await remoteFs.createNew(item.uri, type);
-        this.dataProvider.refreshNode(item.data);
+    private async createNew(item: FileTreeNodeData | DirectoryTreeNodeData, type: 'file' | 'directory' | 'notebook') {
+        const remoteFs = await this.fsFactory.getOrCreateRemoteFileSystem(item.connection);
+        await remoteFs.createNew(Uri.file(item.path).with({ scheme: item.connection.fileScheme }), type);
+        this.dataProvider.refreshNode(item);
     }
     private async refreshServer(item: ServerNode) {
         this.dataProvider.refreshNode(item.data);
     }
     private async refreshKernelSessions(item: KernelSessionsNode) {
         this.dataProvider.refreshNode(item.data);
     }
-    private async delete(item: FileNode) {
+    private async delete(item: FileTreeNodeData) {
```

Comment:
More changes due to changes in VS Code API (no idea if this change was deliberate, not going to look into that).